src/main/java/io/debezium/connector/cassandra/SchemaProcessor.java

```diff
@@ -5,24 +5,202 @@
+                schemaHolder.removeTable(new KeyspaceTable(table));
+
+                final String ksName = table.getKeyspace().getName();
+                final String tableName = table.getName();
+
+                final org.apache.cassandra.schema.KeyspaceMetadata oldKsm = Schema.instance.getKSMetaData(table.getKeyspace().getName());
+
+                if (oldKsm == null) {
+                    throw new IllegalStateException(String.format("KeyspaceMetadata for keyspace %s is not found!", table.getKeyspace().getName()));
+                }
+
+                final ColumnFamilyStore cfs = Keyspace.openWithoutSSTables(ksName).getColumnFamilyStore(tableName);
+
+                if (cfs == null) {
+                    throw new IllegalStateException(String.format("ColumnFamilyStore for %s.%s is not found!", table.getKeyspace(), table.getName()));
+                }
+
+                // make sure all the indexes are dropped, or else.
+                cfs.indexManager.markAllIndexesRemoved();
+
+                // reinitialize the keyspace.
+                final CFMetaData cfm = oldKsm.tables.get(tableName).get();
+                final org.apache.cassandra.schema.KeyspaceMetadata newKsm = oldKsm.withSwapped(oldKsm.tables.without(tableName));
+
+                Schema.instance.unload(cfm);
+                Schema.instance.setKeyspaceMetadata(newKsm);
+
+                logger.info("removed table {}", table.asCQLQuery());
+            }
+
+            @Override
+            public void onTableChanged(final TableMetadata newTableMetadata, final TableMetadata oldTableMetaData) {
+                logger.info(String.format("Detected alternation in schema of %s.%s (previous cdc = %s, current cdc = %s)",
+                        newTableMetadata.getKeyspace().getName(),
+                        newTableMetadata.getName(),
+                        oldTableMetaData.getOptions().isCDC(),
+                        newTableMetadata.getOptions().isCDC()));
+
+                schemaHolder.addTable(new KeyspaceTable(newTableMetadata),
+                        new SchemaHolder.KeyValueSchema(kafkaTopicPrefix, newTableMetadata, sourceInfoStructMaker));
+
+                final CFMetaData rawNewMetadata = CFMetaData.compile(newTableMetadata.asCQLQuery(),
+                        newTableMetadata.getKeyspace().getName());
+
+                final CFMetaData oldMetadata = Schema.instance.getCFMetaData(oldTableMetaData.getKeyspace().getName(), oldTableMetaData.getName());
+
+                // we need to copy because CFMetaData.compile will generate new cfId which would not match id of old metadata
+                final CFMetaData newMetadata = rawNewMetadata.copy(oldMetadata.cfId);
+                oldMetadata.apply(newMetadata);
+
+                logger.info("changed table {}", newTableMetadata.asCQLQuery());
+            }
+        };
+    }
```

Comment:
I'm concerned about the complicated logic and codes involved in these handler functions above for manually updating Cassandra schema.instance or metadata:
- Is there any wrapped-up function in Cassandra library that can be used here directly?
- How much test have you done to ensure the correctness of the manual codes so far?
- Any expensive operation involved in these functions? Like: 
```
 - Keyspace.open(newCFMetaData.ksName).initCf(newCFMetaData, false);
 - Keyspace.openWithoutSSTables(ksName)
```